source/admin_manual/migrating_mailboxes.rst

```diff
@@ -0,0 +1,314 @@
+     inet_listener {
+       port = 12354
+     }
+  }
+
+  doveadm_password = supersecret
+
+Then put doveadm password to new server::
+
+  doveadm_password = supersecret
+
+Now you are ready to migrate.
+
+Executing migration
+-------------------
+
+If you want to avoid any changes to your source system, unidirectional synchronization is recommended.
+
+To migrate users, use::
+
+  doveadm backup -Ru username tcp:host:port
+
+If you are experiencing problems, run::
+
+  doveadm -D backup -Ru username tcp:host:port
+
+This will enable debug logging.
+
+You can run the command again to perform incremental updates.
+
+Note that Public and Shared namespaces are synchronized automatically (see caveats).
+
+It is also possible to do two-way migration if you want to::
+
+  doveadm backup -Ru username tcp:host:port
+  doveadm sync -u username tcp:host:port
+
+This allows you to do more seamless switchover. Test first!
+
+Caveats
+-------
+
+ * Migrating shared namespaces is difficult prior v2.3.15+. To migrate Shared namespaces, you need to use -n Shared -n Shared/Mailbox.
+ * If you are using ACLs, some ACL types can cause problems. Especially if you have ACLs that only permit writing to folder, but not reading it. This applies to shared folders only.
+ * Migration can cause unexpectedly high loads on the source system, beware. There is no way to throttle the synchronization.
+ * Avoid accessing the target mailboxes before the first sync. At best this causes warnings about GUID or UIDVALIDITY changes, and at worst it fails the sync entirely.
+
+.. _migrating_mailboxes_imapc:
+
+Migrating mailboxes over IMAP
+=============================
+
+When migrating mails over IMAP, you need to have valid credentials to the source system. You can either use master password, master user or individual user authentication.
+This depends entirely on the source system. See :ref:`authentication-master_users` for more details.
+
+Preparations
+------------
+
+Configure authentication on the source system to match your preference. Ensure both source and target system agree on usernames.
+
+Configure IMAP client on the target system.
+
+Common settings::
+
+  # these are supported by standard adhering servers
+  imapc_features = rfc822.size fetch-headers
+
+  # If the old IMAP server uses INBOX. namespace prefix, set:
+  #imapc_list_prefix = INBOX
+
+  # Remote hostname
+  imapc_host = remotehost
+
+  ## if you are using TLS
+  imapc_ssl = imaps
+  imapc_port = 993
+
+  # if you are using StartTLS
+  imapc_ssl = starttls
+  imapc_port = 143
+
+  ## these default to system
+  ssl_client_ca_dir = /etc/ssl/certs
+  # or
+  ssl_client_ca_file = /etc/ssl/ca-certificates.pem
+
+Master password::
+
+  imapc_user = %u
+  imapc_password = supersecret
+
+Master user::
+
+  imapc_user = %u
+  imapc_master_user = master-user
+  imapc_password = master-password
+
+Individual password::
+
+  imapc_user = %u
+
+  doveadm -o imapc_password=password sync -1Ru user imapc:
```

Comment:
doveadm backup better to recommend than doveadm sync -1

Reply:
always? even for incrementals?

Reply:
yeah, backup works fine for incrementals as well. The only question is whether it's wanted that destination changes are preserved or not.